src/archive_open.c

```diff
@@ -34,7 +38,7 @@ static int framed_stdin_open(struct archive *a, void *client_data)
 {
     struct fwup_archive_data *ad = (struct fwup_archive_data *) client_data;
 
-#if defined(__CYGWIN__) || defined(_WIN32)
+#ifdef __MINGW32__
```

Comment:
`_WIN32` would be the better one to use since it more generically addresses the issue of why `setmode` is needed. I don't think we can give up on building on Windows yet. My view is to revert the change to the original line. I'll go through the other uses of `__MINGW32__`, but I suspect that they're more appropriately `_WIN32` issues rather than exclusive to `__MINGW32__`.